cmake/modules/AddSwift.cmake

```diff
@@ -1882,6 +1886,16 @@ function(add_swift_target_library name)
        NOT ${name} STREQUAL swiftRemoteMirror)
       list(APPEND swiftlib_link_flags_all "-Wl,-z,defs")
     endif()
+    # Setting back linker flags which are not supported when making Android build on macOS cross-compile host.
+    if(SWIFTLIB_SHARED)
+      if(sdk IN_LIST SWIFT_APPLE_PLATFORMS)
+        list(APPEND swiftlib_link_flags_all "-dynamiclib -Wl,-headerpad_max_install_names")
+      elseif(sdk STREQUAL ANDROID)
+        list(APPEND swiftlib_link_flags_all "-shared")
```

Comment:
What about static builds on Android?

Reply:
Generated ninja rules for static libs seems OK. See default Cmake flags for macOS Host here: `/PATH_TO_HOMEBREW/Cellar/cmake/3.14.5/share/cmake/Modules/Platform/Apple-Clang.cmake`

Reply:
There will be another PR addressed wrong install paths for `libswiftReflection.a` and all shared libs. But that PR will not be addressed `wrong ninja rules generation` in comparison to this one.